mantidproject/mantid
# Add support for masking workspaces to MaskDetectorsIf

**Description of work.**

This PR adds the option to mask workspaces in the `MaskDetectorsIf` algorithm. No changes to the existing functionality to work with calfiles has been altered; one merely has the option to omit the *InputCalFile* and *OutputCalFile* properties while providing *OutputWorkspace*.

**Report to:** [nobody].

**To test:**

Try masking a workspace using different modes and operators. If you have access to calfiles, you can also check that they still work as expected.

Fixes #23414.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
